test/serializers/root_test.rb

```diff
@@ -0,0 +1,19 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class RootTest < Minitest::Test
+
+      def setup
+        @profile = Profile.new({ name: 'Name 1', description: 'Description 1', comments: 'Comments 1' })
+        @profile_serializer = ProfileSerializer.new(@post, {root: 'smth'})
```

Comment:
These tests are testing the value of Serializer#json_key, but the issue in #986 was

```ruby
render json: @current_user, root: 'resource', adapter: :json
```

Which uses the full stack of something like `JsonAdapter.create(ProfileSerializer.new(@post, root: 'smth')).serializable_hash`.  I'm not sure you just want to test `ProfileSerializer.new(@post, root: 'smth').json_key`.

That said, the behavior you desire was specifically removed from AMS in #958 

```diff
# controller
-    ADAPTER_OPTION_KEYS = [:include, :fields, :root, :adapter]
+    ADAPTER_OPTION_KEYS = [:include, :fields, :adapter]
# serializer
     def initialize(object, options = {})

-      @root       = options[:root] || (self.class._root ? self.class.root_name : false)
+      @root       = options[:root]

    def json_key
-      if root == true || root.nil?
-        self.class.root_name
-      else
-        root
-      end
+      self.class.root_name
     end
 # adapter

       def root
-        @options.fetch(:root) { serializer.json_key }
+        serializer.json_key.to_sym if serializer.json_key
       end
 
       def include_meta(json)
-        json[meta_key] = meta if meta && root
+        json[meta_key] = meta if meta
         json
       end
```

I brought up documenting these changes in #966 but it hasn't been merged/finalized yet